src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -343,34 +343,11 @@ private static IntPtr GetSslMethod(SslProtocols protocols)
 
         private static int VerifyClientCertificate(int preverify_ok, IntPtr x509_ctx_ptr)
         {
-            using (SafeX509StoreCtxHandle storeHandle = new SafeX509StoreCtxHandle(x509_ctx_ptr, false))
-            {
-                using (var chain = new X509Chain())
-                {
-                    chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
-                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
-
-                    using (SafeX509StackHandle chainStack = Crypto.X509StoreCtxGetChain(storeHandle))
-                    {
-                        if (chainStack.IsInvalid)
-                        {
-                            Debug.Fail("Invalid chain stack handle");
-                            return 0;
-                        }
-
-                        IntPtr certPtr = Crypto.GetX509StackField(chainStack, 0);
-                        if (IntPtr.Zero == certPtr)
-                        {
-                            return 0;
-                        }
-
-                        using (X509Certificate2 cert = new X509Certificate2(certPtr))
-                        {
-                            return chain.Build(cert) ? 1 : 0;
-                        }
-                    }
-                }
-            }
+            // Full validation is handled after the handshake in VerifyCertificateProperties and the
+            // user callback.  It's also up to those handlers to decide if a null certificate
+            // is appropriate.  So just return 1 and tell OpenSSL that the cert is acceptable,
+            // we'll process it after the handshake finishes.
+            return 1;
```

Comment:
Nit: could we put the 1 in a named constant, to help explain it's meaning?  I had to look up in the OpenSSL docs what this function is supposed to return.  Maybe:
```C#
const int PreverifyOk = 1;
return PreverifyOk;
```
or potentially better have this defined in the interop files and verified with a static assert on the native side to match preverify_ok.

(I realize you didn't introduce this here, but we might as well clean it up as we go ;)